File: restdoc-rpc-client/restdoc-dubbo-client/src/main/java/restdoc/client/dubbo/EnvConfiguration.java
package restdoc.client.dubbo;

import org.springframework.boot.context.properties.EnableConfigurationProperties;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import restdoc.client.api.AgentConfigurationProperties;
import restdoc.client.api.AgentImpl;

@EnableConfigurationProperties(value = {AgentConfigurationProperties.class})
@Configuration
public class EnvConfiguration {

    @Bean(name = "dubboAgentImpl")
    AgentImpl agentImpl(AgentConfigurationProperties properties) {
        return new AgentImpl(properties);
    }
}

File: restdoc-rpc-client/restdoc-dubbo-client/src/main/kotlin/restdoc/client/dubbo/EnvConfiguration.kt
package restdoc.client.dubbo

import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import restdoc.client.api.AgentConfigurationProperties
import restdoc.client.api.AgentImpl

@EnableConfigurationProperties(value = [AgentConfigurationProperties::class])
@Configuration
open class EnvConfiguration {

    @Bean(name = ["dubboAgentImpl"])
    open fun agentImpl(properties: AgentConfigurationProperties) = AgentImpl(properties)
}